cas-server-core/src/main/java/org/jasig/cas/services/DefaultRegisteredServiceAuthorizationStrategy.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.services;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Sets;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.jasig.cas.authentication.principal.Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.validation.constraints.NotNull;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This is {@link DefaultRegisteredServiceAuthorizationStrategy}
+ * that allows the following rules:
+ *
+ * <ul>
+ *     <li>A service may be disallowed to use CAS for authentication</li>
+ *     <li>A service may be disallowed to take part in CAS single sign-on such that
+ *     presentation of credentials would always be required.</li>
+ *     <li>A service may be prohibited from receiving a service ticket
+ *     if the existing principal attributes don't contain the required attributes
+ *     that otherwise grant access to the service.</li>
+ * </ul>
+ *
+ * @author Misagh Moayyed mmoayyed@unicon.net
+ * @since 4.1
+ */
+public class DefaultRegisteredServiceAuthorizationStrategy implements RegisteredServiceAuthorizationStrategy {
+
+    private static final long serialVersionUID = 1245279151345635245L;
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    /** Is the service allowed at all? **/
+    private boolean enabled = true;
```

Comment:
Sure, point understood. At the risk of repeating myself :) I would add that the presented authorization strategy is more of a conglomerate collection of rules that would disable user access to CAS and thus the app. It combines configuration that can be applied to CAS and retrieves configuration that lives outside CAS for authorization purposes to compose all of that functionality in one spot. It's just a matter of, IMO, how you read the functionality out loud in terms of a use case. We could, as you suggest, separate the two (as the previous addon also did and focused only on the user-specific part). 

I certainly do agree that it would be weird to use "enabled" + "user-attribute enabled/authorized" at the same time. That would be wrong and impractical. Does this pull encourage that behavior? Not sure. Maybe we can better addressed by documentation and sample use cases? perhaps :)

Let me ping @leleuj and @serac. What's your take here?